### ISSUE TITLE
Data: do we have validators that are accepting blocks without verifying them?

### ISSUE THREAD

**Issue by JuanBQti** [cometbft/cometbft#782]
## Feature Request

Collect data to check if there are validators that accept blocks without verifying them.
Idea: check if they accept blocks when they should reject them. 
(any other feature could also be used to measure the behavior of validators, as described in the problem definition).

### Summary

We need to collect data to check if there are validators that accept/reject blocks without verifying them.

### Problem Definition

Let's imagine a validator who is not Byzantine. That is, the validator wants to reach consensus and does not want to attack the protocol. The validator also wants to maximize his profits. Suppose this validator has very low voting power, say 0.1%.

Suppose also that the validator anticipates or believes that all other validators will verify the block and vote based on such verification. Our validator knows that it makes no difference whether he accepts the block or not. The voting power is so low that it will not change the decision. The block will be produced (or not) by the votes of others. Hence, why does our validator need to verify? **He could just free-ride: vote without verifying**. 

Free-riding is profitable: Since the validator is paid when he votes a valid block, but not punished when he votes an invalid block, our validator can vote without verifying (remember that it is not possible to see if a validator verifies or not. We only observe the votes). This is cheaper for the validator. The risk of being punished for producing an invalid block is very low since the other validators verify the block.

**Why is important?** If this is the only validator deviating, the consensus is reached anyway. However, we are paying a validator for not doing his job. Is he the only one?
The risk for the protocol might be important if the number of deviating validators is larger. (new vector attacks? reputation effects? no consensus?)

There is a recent game-theoretical paper studying this issue. The paper models PoS à la Tendermint. The good news is that the paper shows that consensus can be reached under some conditions. See my summary and personal notes in:
https://github.com/informalsystems/economics-analysis/issues/12

Note that the previous description does not include the risk of being in a double-signing case. The cost of being punished in such a case is large. However, from historical data, the probability of occurrence is low and, hence, the expected cost might be also low. Not sure about this part. It could be a sufficient reason to avoid the free-riding strategy. In any case, observing and understanding the validators' behavior might be valuable.

### Proposal

Collect data to check whether or not there are validators following that free-rider strategy.

**Comment by cason** [comment]
Related: https://github.com/tendermint/tendermint/issues/9504

In fact, it can be worse. Once a validator receives a vote for a `BlockID`, it can "support" this vote without even having received the full `Block` the `BlockID` points to.

**Comment by JuanBQti** [comment]
Good point. 
Empty blocks, are they accepted to avoid delays? or are they rejected? If it is the last one, then we could just check if our free-rider is voting empty blocks

**Comment by cason** [comment]
By "empty", do you mean without transactions? Yes, nodes should accept them.

**Comment by JuanBQti** [comment]
yes, that is why I deleted the comment. I just checked in mintscan and we produce empty blocks

**Comment by JuanBQti** [comment]
I agree with your previous comment about the BlockID...

**Comment by lasarojc** [comment]
Is anyone rewarded for reporting misbehavior from the ATOM slashed?

**Comment by JuanBQti** [comment]
I do not think so

**Comment by JuanBQti** [comment]
can we check if some nodes are proposing empty blocks each time that they are proposers?

**Comment by cason** [comment]
Do you mean, manually or write a script for that? Yes. We can have the list of blocks per public key (validator) and check the number of transactions each one has. One can do that using some mint scan or other websites.

This is not a check we do when validating a block.

**Comment by JuanBQti** [comment]
well, manually, as you said, I can go to mintscan.io and see which blocks are empty. We can also see the proposer there. But I can only observe the last 50 blocks or so.

Maybe we need to check more data points. a couple of days (~30,000 blocks)? one week (~100k)?

**Comment by cason** [comment]
@JuanBQti, I think the concern of this issue is more related to running in production networks than to CometBFT.

Any objection to closing this issue?

**Comment by sergio-mena** [comment]
Hi, since this an enhancement, we don't need to close it. It shouldn't count as a bug
